include/lomse_gm_basic.h

```diff
@@ -288,7 +288,7 @@ class GmoObj
     void set_left(LUnits xLeft);
     void set_top(LUnits yTop);
     virtual void shift_origin(const USize& shift);
-    void shift_origin(LUnits x, LUnits y);
+    virtual void shift_origin(LUnits x, LUnits y);
```

Comment:
Another possible option I see is to leave this function non-virtual and just make it call `shift_origin(USize(x, y))`. Then implementations of these two functions will always behave the same way, without the need to override both of them.

Reply:
Thank you. You are right, current code duplicates the implementation with the risk of introducing different behavior. I will change this.

Reply:
Finally, it has to be defined as virtual. Otherwise, if the `GmoCompositeShape` is managed as a pointer to a shape (`GmoShape`), then the base class method is invoked instead of the shadowing method, which is logical. So if method with two parameters is used, i has to be defined as virtual. I left the override, but reviewed implementations to avoid repeating the code.

Reply:
We don't need to add shadowing methods if the method in the base class is defined by calling the virtual method, it is enough just to override the virtual method in inherited classes. My proposal is just to remove `shift_origin(LUnits x, LUnits y)` from all classes except the base class (`GmoObj` in this case) and re-define only `shift_origin(const USize&)` method in inherited classes. Then if the first method is defined in the base class like this:
```
void shift_origin(LUnits x, LUnits y) { shift_origin(USize(x, y)); }
```
then the virtual `shift_origin` flavor will always be called, always with the implementation suitable for the corresponding object type.

Reply:
Thank you. The approach you suggests is correct. I've tried but after several test I'm not able to make my code work. I don't know were is the problem, a bug in my compiler `g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0` or a definition problem in my code, but if I only define in base class `GmoObj`:
```
void shift_origin(LUnits x, LUnits y) { shift_origin(USize(x, y)); }
```
the compiler does not recognizes this method as an inherited method in GmoShapeRest class (derived from GmoCompositeShape) and I get two errors:

```
 /datos/cecilio/lm/projects/lomseapp/lomse/trunk/src/graphic_model/layouters/lomse_noterests_collisions_fixer.cpp: In member function ‘void lomse::NoterestsCollisionsFixer::move_rest(size_t, lomse::LUnits)’:
/datos/cecilio/lm/projects/lomseapp/lomse/trunk/src/graphic_model/layouters/lomse_noterests_collisions_fixer.cpp:628:42: error: no matching function for call to ‘lomse::GmoShapeRest::shift_origin(float, lomse::LUnits&)’
     pRestShape->shift_origin(0.0f, yShift);
                                          ^
In file included from ../../lomse/trunk/include/private/lomse_internal_model_p.h:52:0,
                 from ../../lomse/trunk/include/lomse_internal_model_defs.h:35,
                 from ../../lomse/trunk/include/lomse_internal_model.h:33,
                 from /datos/cecilio/lm/projects/lomseapp/lomse/trunk/src/graphic_model/layouters/lomse_noterests_collisions_fixer.cpp:33:
../../lomse/trunk/include/lomse_shape_base.h:156:18: note: candidate: virtual void lomse::GmoCompositeShape::shift_origin(const USize&)
     virtual void shift_origin(const USize& shift) override;
                  ^~~~~~~~~~~~
../../lomse/trunk/include/lomse_shape_base.h:156:18: note:   candidate expects 1 argument, 2 provided
/datos/cecilio/lm/projects/lomseapp/lomse/trunk/src/graphic_model/layouters/lomse_noterests_collisions_fixer.cpp: In member function ‘void lomse::NoterestsCollisionsFixer::move_notehead(lomse::GmoShapeNote*, lomse::LUnits)’:
/datos/cecilio/lm/projects/lomseapp/lomse/trunk/src/graphic_model/layouters/lomse_noterests_collisions_fixer.cpp:1001:46: error: no matching function for call to ‘lomse::GmoShapeNote::shift_origin(lomse::LUnits&, float)’
         pShapeNote->shift_origin(xShift, 0.0f); //shift all to right
                                              ^
In file included from ../../lomse/trunk/include/private/lomse_internal_model_p.h:52:0,
                 from ../../lomse/trunk/include/lomse_internal_model_defs.h:35,
                 from ../../lomse/trunk/include/lomse_internal_model.h:33,
                 from /datos/cecilio/lm/projects/lomseapp/lomse/trunk/src/graphic_model/layouters/lomse_noterests_collisions_fixer.cpp:33:
../../lomse/trunk/include/lomse_shape_base.h:156:18: note: candidate: virtual void lomse::GmoCompositeShape::shift_origin(const USize&)
     virtual void shift_origin(const USize& shift) override;
                  ^~~~~~~~~~~~
../../lomse/trunk/include/lomse_shape_base.h:156:18: note:   candidate expects 1 argument, 2 provided

```

So after several tests and changes without solving the issue, as the method with two parameters is only used in tree or four places and, at the end, in base class, the USize object will be created to invoke the method doing the work, I've opted for not lossing more time and definitly removing the method with to arguments.

Reply:
Yes, if one of those `shift_origin`s can be removed that would be even simpler.

Otherwise this PR looks good for me, thanks a lot!